src/ng/parse.js

```diff
@@ -1931,40 +1934,58 @@ function $ParseProvider() {
       return unwatch;
     }
 
+    function addWatchDelegate(parsedExpression) {
+      if (parsedExpression.constant) {
+        parsedExpression.$$watchDelegate = constantWatchDelegate;
+      } else if (parsedExpression.oneTime) {
+        parsedExpression.$$watchDelegate = oneTimeWatchDelegate;
+      } else if (parsedExpression.inputs) {
+        parsedExpression.$$watchDelegate = inputsWatchDelegate;
+      }
+
+      return parsedExpression;
+    }
+
+    function chainInterceptors(first, second) {
+      function chainedInterceptor(value, scope, locals) {
+        return second(first(value, scope, locals), scope, locals);
```

Comment:
This is equivalent to today. The interceptor [is passed the scope and locals](https://github.com/angular/angular.js/blob/v1.6.5/src/ng/parse.js#L1963). Then if it is double-intercepted then it will [pass them through](https://github.com/angular/angular.js/blob/v1.6.5/src/ng/parse.js#L1962) until reaching the original expression.

I would be tempted to remove these args from the interceptor though, and only pass them to the original wrapped expression. Probably in a separate PR though...